parser/src/main/java/io/specmesh/apiparser/model/Message.java

```diff
@@ -70,6 +77,50 @@ public class Message {
      * @return the location of the schema
      */
     public String schemaRef() {
-        return (String) payload.get("$ref");
+        final Object o = payload.get("$ref");
+        if (o == null) {
+            throw new APIParserException("$ref lookup not supported on this spec format");
+        }
+        return (String) o;
+    }
+
+    /**
+     * return properties: value: type: long value: type: $ref: "MessageValue.avsc". # avro-type (or
+     * json or proto)
+     *
+     * @return long or schema ref
+     */
+    public String valueType() {
```

Comment:
Given data that looks like:

```yaml
message:
  payload:
    properties:
      key: 
         type: 
            long  # in build kafka-type
      value:
        type:
          $ref: "MessageValue.avsc". # avro-type
```

It loos like `valueType()` would return `"$ref: "MessageValue.avsc"."`... which is a bit weird, no?

I would have expected the return type of keyType() and valueType() to be a polymorphic type, e.g. `TypeInfo`, with two implementations: `KafkaType` (which could be an enum of supported types), and `SchemaTypeRef` which is a reference to an external schema file, and has an accessor the access the path.